pennylane/measurements/mid_measure.py

```diff
@@ -218,6 +232,18 @@ def _merge(self, other: "MeasurementValue"):
         merged_measurement_ids = list(set(self.measurement_ids).union(set(other.measurement_ids)))
         merged_measurement_ids.sort()
 
+        merged_measurements = []
+        for item in self.measurements + other.measurements:
```

Comment:
What is the difference between this loop and `list(set(self.measurements + other.measurements))`?

Reply:
We are going to be changing the definition for `__eq__` and `__hash__` for operators and measurement processes in the near future.  Right now those depend on the "identity", but we are going to make them depend on the contents.

Reply:
The hash of measurement processes will be udpated soon. Using a set would work right now, but the behaviour will change later. Although as I say this, the hash for `MidMeasureMP` could be updated so that it depends on the id, in which case the hash for different objects would still be different, so this change should work. I'll make the change.

Reply:
@albi3ro